### PR TITLE
fix track desc

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Review by RZR-UA — APPROVED** [review]

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2128

File: map/src/test/validateTrackDesc.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+test('Validate desc', () => {
+    expect(prepareDesc(testData[0])).toBe(results[0]);
+});
+test('Validate desc2', () => {
+    expect(prepareDesc(testData[1])).toBe(results[1]);
+});
+test('Validate desc3', () => {
+    expect(prepareDesc(testData[2])).toBe(results[2]);
+});
```
Comment (RZR-UA) [inline map/src/test/validateTrackDesc.test.js:57]: yarn lint failed - should be improved to lint jest tests

File: map/src/context/TracksManager.js [hunk 2]
```diff
@@ -1212,6 +1213,10 @@ export const getGpxTime = (f, reverse = false) => {
     return 0;
 };
 
+export function prepareDesc(trackDesc) {
+    return anchorme(trackDesc.replace(/\n/g, '<br />'));
+}
```
Comment (RZR-UA) [inline map/src/context/TracksManager.js:1218]: Please add check trackDesc.

Selenium tests failed (no .replace in trackDesc).

Also, Jest-test should have more input data such as: [null, '', undefined, ...]

File: tests/selenium/src/tests/50-infoblock-desc.mjs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import actionIdleWait from '../actions/actionIdleWait.mjs';
+import { driver } from '../options.mjs';
+
+const TEST_GPX_FILE = 'test-infoblock-desc.gpx';
+const DESC = '1 test\n' + '2 test\n' + '3 test\n' + 'www.osmand.net';
+export default async function test() {
+    await actionOpenMap();
+
+    await clickBy(By.id('se-show-main-menu'), { optional: true });
+    await clickBy(By.id('se-show-menu-tracks'), { optional: true });
```
Comment (RZR-UA) [inline tests/selenium/src/tests/50-infoblock-desc.mjs:14]: you can remove this 2 lines if you switch upload to action: actionUploadGpx() (see test example 20-local-to-cloud)

File: tests/selenium/src/tests/50-infoblock-desc.mjs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    await actionIdleWait();
+    await matchTextBy(By.id('se-infoblock-desc'), DESC);
+    await driver.findElement(By.xpath("//a[@href='http://www.osmand.net']")).click();
+}
+
+const uploader = async () => {
+    const element = await waitBy(By.id('se-upload-gpx'));
+    await element.sendKeys(resolve('gpx', TEST_GPX_FILE));
+    return true;
+};
```
Comment (RZR-UA) [inline tests/selenium/src/tests/50-infoblock-desc.mjs:26]: Better use actionUploadGpx({ mask: TEST_GPX_FILE })

File: tests/selenium/src/tests/50-infoblock-desc.mjs [hunk 5]
```diff
@@ -16,7 +16,7 @@ export default async function test() {
     await enclose(uploader, { tag: 'upload gpx' });
     await actionIdleWait();
     await matchTextBy(By.id('se-infoblock-desc'), DESC);
-    await driver.findElement(By.xpath("//a[@href='http://www.osmand.net']")).click();
+    await driver.findElement(By.xpath("//a[@href='http://www.osmand.net']"));
```
Comment (RZR-UA) [inline tests/selenium/src/tests/50-infoblock-desc.mjs:19]: please use lib waitBy()

for simple tests we should prefer library functions, not direct driver() calls
